CTS 0.6.0 Beta Docs: Health endpoint and service registration (#12967)
hashicorp/consul: website/content/docs/release-notes/consul-terraform-sync/v0_6_0_beta.mdx

### Response Statuses

| Status | Reason                                               |
| ------ | ---------------------------------------------------- |
| 200    | CTS is healthy                                       |

### Response Fields

| Name         | Type   | Description                                                                        |
| ------------ | ------ | ---------------------------------------------------------------------------------- |
| `request_id` | string | The ID of the request. Used for auditing and debugging purposes.                   |

### Example

The following request makes a `GET` call to the `health` endpoint:

```shell-session
$ curl --request GET \
  localhost:8558/v1/health
```

Response:

```json
{
  "request_id": "b7559ab0-5111-381b-367a-0dfb7e216d41"
}
```

## CTS Service Registration With Consul
By default, CTS automatically registers itself as a service in Consul with a health check. The default health check is an HTTP check to the [health API endpoint](#health-endpoint). CTS deregisters itself with Consul when it is stopped.

If CTS is unable to register with Consul, then it will log the error and continue without exiting.

> ~> If ACLs are enabled in Consul, then the [Consul token](/docs/nia/configuration#token) must have the policy `service:write` for service registration to succeed.

Reviewer: ```suggestion
If ACLs are enabled in Consul, then the [Consul token](/docs/nia/configuration#token) must have the policy `service:write` for service registration to succeed.
```
I think this is just normal information, yes? No reason to render as a callout, is there?

Reviewer 2: I called it out because CTS previously didn't have this requirement and the registration feature is enabled by default. I could add that information if that would make it more clear!

Reviewer 2: Or in this case, does it still make sense not to make it a callout?

Reviewer: I think it's fine as part of the regular content for this section in plain text. I believe that we want people to enable ACLs in a production environment, so I don't know that this would constitute exceptional or orthogonal information.

Reviewer 2: Okay, I'll modify this to not be a callout then! I'll also add a (plain text) note specifying how this was not previously a requirement so that users may need to modify their token, though. Since these are more or less detailed release notes for the beta, I do want to make sure they understand that a token that worked for them in 0.5 may not work for this service registration feature in the 0.6.0-beta.

### Configuration

> To support registration, CTS now has an `id` configuration that is a global configuration and a `service_registration` configuration that is part of the `consul` configuration. All configurations are optional.

Reviewer: ```suggestion
Specify the `id` parameter to assign instances of CTS an ID. Configure the  `service_registration` block in the `consul` configuration to enable CTS registration. All configurations are optional.
```
Right?

Reviewer 2: ```suggestion
Specify the global configuration `id` to assign instances of CTS an ID, which is used as the service ID. Configure the `service_registration` block in the `consul` configuration to configure the registration details. All configurations are optional.
```

I made some modifications, let me know what you think! You don't have to configure `service_registration` to enable it since it's on by default, and I really wanted to keep global configuration since that's how we refer to these top-level configurations ([reference](https://www.consul.io/docs/nia/configuration#global-config-options)).

Reviewer 2: @trujillo-adam Updated it to the following:
> Specify the global configuration `id` to assign instances of CTS an ID, which is used as the service ID for CTS if service registration is enabled. Configure the `service_registration` block in the `consul` configuration to configure the registration details. All configurations are optional.

Please let me know if you have any concerns, I'll be merging this EOD today otherwise so that it's ready for our beta release tomorrow.

```hcl
id = "cts-01"

consul {
  service_registration {
    enabled = true
    service_name = "cts"
    namespace = "ns-1"
    default_check {
      enabled = true
      address = "http://172.22.0.2:8558"
    }
  }
}
```

> The options for the global configuration `id` are:

Reviewer: ```suggestion
The following table describes the options for the `id` global configuration:
```
Better to write with complete sentences in prose.

| Name            | Required | Type    | Description                     |Default                  |
| --------------- | -------- | ------- | ------------------------------- | ----------------------- |
| `id ` | Optional | string | The ID of the CTS instance. Used as the service ID for CTS if service registration is enabled. | Generated ID with the format `cts-<uuid>` |

> The configuration options for `consul.service_registration` are:

Reviewer: ```suggestion
The following table describes configuration options for `consul.service_registration`:
```

| Name            | Required | Type    | Description                     |Default                  |
| --------------- | -------- | ------- | ------------------------------- | ----------------------- |

> | `enabled ` | Optional | boolean | Whether CTS should automatically register itself as a service with Consul. | `true` |

Reviewer: ```suggestion
| `enabled ` | Optional | boolean | Enables CTS to register itself as a service with Consul. | `true` |
```

> | `service_name` | Optional | string | The service name for CTS. | "Consul-Terraform-Sync" |

Reviewer: ```suggestion
| `service_name` | Optional | string | The service name for CTS. | `Consul-Terraform-Sync` |
```

| `namespace` | Optional | string |  <EnterpriseAlert inline /> The namespace to register CTS in. | In order of precedence: <br/> 1. Inferred from the CTS ACL token <br/> 2. The `default` namespace. |

> | `default_check.enabled` | Optional | boolean | Whether CTS should create the default health check. | `true` |

Reviewer: ```suggestion
| `default_check.enabled` | Optional | boolean | Enables CTS to create the default health check. | `true` |
```

> | `default_check.address` | Optional | string | The address to use for the default HTTP health check. Needs to include the scheme (http/https) and the port (if necessary). | `http://localhost:<port>` or `https://localhost:<port>`. Determined from the [port configuration](/docs/nia/configuration#port) and whether [TLS is enabled](/docs/nia/configuration#enabled-2) on the CTS API. |

Reviewer: ```suggestion
| `default_check.address` | Optional | string | The address to use for the default HTTP health check. Needs to include the scheme (`http`/`https`) and the port (if necessary). | `http://localhost:<port>` or `https://localhost:<port>`. Determined from the [port configuration](/docs/nia/configuration#port) and whether [TLS is enabled](/docs/nia/configuration#enabled-2) on the CTS API. |
```
